### ISSUE TITLE
WCAG2ICT Project: Add 2.5.2 Pointer Cancellation

### ISSUE THREAD

**Issue by maryjom** [w3c/wcag2ict#32]
- [x] Add [2.5.2 Pointer Cancellation](https://www.w3.org/TR/WCAG22/#pointer-cancellation) section (from WCAG 2.1)
- [x] Ensure new definitions of "down-event" and "up-event" are reviewed for possible interpretation (See Issue #22) 
- [ ] Add in the [Understanding 2.5.2 Pointer Cancellation](https://www.w3.org/WAI/WCAG21/Understanding/pointer-cancellation.html) Intent and Benefits sections
- [x] Consider and write up whether there are any notes needed for application of this SC to non-web technologies

**Comment by maryjom** [comment]
Current definition of "down-event" in WCAG 2.2:

>**down-event**
>platform event that occurs when the trigger stimulus of a pointer is depressed
>
>The down-event may have different names on different platforms, such as "touchstart" or "mousedown".

May want WCAG2ICT verbiage for the explanation to give examples of non-web software names of these events:

**down-event**
This applies directly as written and as described in the WCAG 2.2 glossary.
 
Note: The down-event has different names on different platform software. For example “pointerpressed” or “touchesBegan”.

**Comment by oh185014** [comment]
From [2.5.2 Pointer Cancellation](https://www.w3.org/TR/WCAG22/#pointer-cancellation):

> For [functionality](https://www.w3.org/TR/WCAG22/#dfn-functionality) that can be operated using a [single pointer](https://www.w3.org/TR/WCAG22/#dfn-single-pointer), at least one of the following is true:
> - **No Down-Event:** The [down-event](https://www.w3.org/TR/WCAG22/#dfn-down-event) of the pointer is not used to execute any part of the function;
> - **Abort or Undo:** Completion of the function is on the [up-event](https://www.w3.org/TR/WCAG22/#dfn-up-event), and a [mechanism](https://www.w3.org/TR/WCAG22/#dfn-mechanism) is available to abort the function before completion or to undo the function after completion;
> - **Up Reversal:** The up-event reverses any outcome of the preceding down-event;
> - **Essential:** Completing the function on the down-event is [essential](https://www.w3.org/TR/WCAG22/#dfn-essential).
> 
> **NOTE:** Functions that emulate a keyboard or numeric keypad key press are considered essential.
> 
> **NOTE:** This requirement applies to web content that interprets pointer actions (i.e. this does not apply to actions that are required to operate the user agent or assistive technology).

**Comment by maryjom** [comment]
@oh185014 If this helps, I think the only WCAG2ICT considerations for this would be for the 2nd note because it's talking about "web content" where I think the term should be just "content" with a link to the [WCAG2ICT definition of content](https://w3c.github.io/wcag2ict/#content-on-and-off-the-web).  There's also the use of "user agent or assistive technology" where you could add "platform software" to that.

**Comment by oh185014** [comment]
Thanks @maryjom!

I agree, beyond the word switches I do not believe any additional notes are needed. This is ready for TF to review.

**Comment by maryjom** [comment]
@oh185014 Sorry to churn this, as I think I just cursory looked a this one without enough thought. I just realized the word substitutions I had suggested and where this might make sense as a requirement for non-web software (based on comments on the gestures SC). I think we can look to the EN 301 549's treatment for this to help us consider documents and non-web software as separate cases. This could be a good discussion topic in the meeting on March 8, if there's time.

#### Non-web software
Notes in the EN 301 549 for **Software success criterion: 2.5.2 Pointer cancellation** are:
>NOTE 1: Functions that emulate a keyboard or numeric keypad key press are considered essential.  
>NOTE 2: This requirement applies to **[(non-web software)]** that interprets pointer actions (i.e. this does not apply to actions that are required to operate the user agent or assistive technology).
>NOTE 3: This success criterion is identical to the WCAG 2.1 Success Criterion 2.5.2 Pointer Cancellation replacing the original WCAG 2.1 note with notes 1 and 2 above.

My detailed analysis:
**User agents:** When the non-web software is a user agent, it might interpret its own pointer actions. So should this have an exemption as Note 2 above states?  I don't think so. 
**Native software (that is not a platform)**: Software that is not a platform shouldn't be responsible actions required to operate the platform software. So that should have this note: 
>NOTE: This requirement applies to **[non-web software that is not a platform]"**. Non-platform software is not responsible for the accessibility of pointer actions that are required to operate the platform or assistive technology.
**Platform software** This requirement should apply, as written, no exemption clause.  
>NOTE: This requirement applies to **[(platform software)]** that interprets pointer actions (i.e. this does not apply to actions that are required to operate assistive technology). 
**Closed functionality software** This requirement should apply, and if there's no pointer support this would be a N/A requirement. There may be some odd issues I don't know about. @samogami or @pday1 may have some input on this.

This could sum up into two notes for non-web software:
NOTE: This requirement applies to **[non-web software]** that interprets pointer actions (i.e. this does not apply to actions that are required to operate assistive technology).
NOTE: Non-web software applications are not responsible for pointer actions required to operate the underlying platform software.

This does not address the EN note 1: "Functions that emulate a keyboard or numeric keypad key press are considered essential. " I'm sure @mapluke can enlighten us on this one. Maybe it's to handle mobile virtual keyboard typing where you can drag your finger and pause on letters to type.

#### Non-web documents
Should this be applied to non-web documents? Can a document be aware of pointer events or be in control of what happens on a down-event vs. up-event? Maybe Figma or AdobeXD have this capability? Maybe others on the task force or in AG WG know more. May need a note similar to what I suggested for 2.5.1 Pointer Gestures in this [comment](https://github.com/w3c/wcag2ict/issues/31#issuecomment-1460438503).

**Comment by mraccess77** [comment]
I agree with the suspicion on-screen keyboards with path based typing is the reason there is a note about keyboard/keyboard functionality being essential.  What's interesting is that the default on iOS for individual on-screen keys is to support activation on the up event - but when using the path based typing to type out a word it reacts on the up event of the predicted word and not each letter.  So it seems like there is still good reason to have individual keys support activation on the up event while offering other solutions to make typing more efficient.

I think a note might be ok on the essential exception if we think there would confusion on how to interpret what is essential or not - but it doesn't seem to change the normative text.

**Comment by samogami** [comment]
"Waking" a device from sleep mode, increasing the display's brightness, and showing additional options on the screen count as [functionality](https://www.w3.org/TR/WCAG22/#dfn-functionality)?  Devices with touch screens that also have power-saving requirements start when the user first touches that device.  
I suggest adding a note.
**Note:** Additional essential functions for non-web software and documents are changing the device to a user-ready state or providing additional contextual options. 

I also question if this needs to be applied to all non-web software.  Windows tabs including the ones in MS Edge and Chrome will change if using a mouse with the down click.  If you use the keyboard you need to select the tab and then use space or enter to change the show tab.

**Comment by olivia-hogan-stark** [comment]
@maryjom Haha no worries at all for “churning this”! Thanks for putting more thought into it. I was unexpectedly OOO end of last week, so sorry I wasn’t at last week’s meeting to discuss. I am at CSUN this week, but would be happy to discuss the following Thursday the 23rd. 

–––

Draft notes and discussion points based on comments so far: 

> NOTE 1: Functions that emulate a keyboard or numeric keypad key press are considered essential.

Discuss @mraccess77 comment:
- Keep normative text
- Do we need an additional note? 

> NOTE 2: Additional essential functions for non-web software and documents are changing the device to a user-ready state or providing additional contextual options.

@samogami suggested this additional note

> NOTE 3: This requirement applies to [non-web software] that interprets pointer actions (i.e. this does not apply to actions that are required to operate assistive technology).
> 
> NOTE 4: Non-web software applications are not responsible for pointer actions required to operate the underlying platform software.

@maryjom suggested note 3 and 4 in response to EN 301 549 Note 2: NOTE 2: This requirement applies [to a document/to non-web software] that interprets pointer actions (i.e. this does not apply to actions that are required to operate the user agent or assistive technology).

**Comment by pday1** [comment]
Closed system: another example where it could cause issues is where a touchscreen is encrypted for security (e.g. entering a personal identification number onscreen). In this case,  the underlying software may not be allowed to know some of these state changes, and therefore we either classify this in the essential functions note, or as another example in Sam's closed functionality examples

**Comment by maryjom** [comment]
Opened Issue #124 to consolidate comments made so far (and incorporate thoughts generated from our brief discussion in the 16 March meeting) into something we might be able to survey.

**Comment by maryjom** [comment]
This version of 2.5.2 was approved by the AG WG on 11 April. Closing.
